boa/src/value/mod.rs

```diff
@@ -456,19 +461,21 @@ impl Value {
     /// Resolve the property in the object and get its value, or undefined if this is not an object or the field doesn't exist
     /// get_field receives a Property from get_prop(). It should then return the `[[Get]]` result value if that's set, otherwise fall back to `[[Value]]`
     /// TODO: this function should use the get Value if its set
-    pub fn get_field<K>(&self, key: K) -> Self
+    pub fn get_field<K>(&self, key: K, context: &mut Context) -> Result<Self>
     where
         K: Into<PropertyKey>,
     {
         let _timer = BoaProfiler::global().start_event("Value::get_field", "value");
         let key = key.into();
         match self.get_property(key) {
             Some(ref desc) => match desc {
-                // TODO: Add accessors
-                PropertyDescriptor::Accessor(_) => Value::undefined(),
-                PropertyDescriptor::Data(desc) => desc.value(),
+                PropertyDescriptor::Accessor(AccessorDescriptor { get: Some(get), .. }) => {
+                    get.call(&self.clone(), &[], context)
+                }
+                PropertyDescriptor::Data(desc) => Ok(desc.value()),
+                _ => Ok(Value::undefined()),
```

Comment:
It would be better if didn't have to define the same implementation as `.get` but call get directly

Reply:
Maybe I could change all the calls to `Value::get/set_field` to to `to_object(..)?.get/set` as you have suggested in #577. Now that context has been added to `get/set_field`, the bulk of the work would be a search and replace. What do you think?

Reply:
> Maybe I could change all the calls to `Value::get/set_field` to `to_object(..)?.get/set` as you have suggested in #577. Now that context has been added to `get/set_field`, the bulk of the work would be a search and replace. What do you think?

It's not what needs to be done `.to_object()` does some other things and would not be spec compliant, the calls to `.get/.set` should be reinforced by the type system as it does not make much sense to have set/get for non-object (`Value` type) values as explained in #577.

we can merge this as it's main purpose was to add property accessors, and replace `.get_field` and `.set_field` builtin function by builtin function, through incremental changes (with many PRs), Allowing `.get_field/set_field` to take context is the first step, the end goal should be to completely remove them, it's really hard otherwise.

Reply:
I should have thought that if it was so easy, it would have already been done. Sorry if that suggestion seemed silly, I am quite a newbie on this code base.